tests/ci/integration/run_libgit2_integration.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -exu
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+
+# SYS_ROOT
+#  |
+#  - SRC_ROOT(aws-lc)
+#  |
+#  - SCRATCH_FOLDER
+#    |
+#    - libgit2
+#    - LIBGIT2_BUILD_FOLDER
+#    - LIBGIT2_INSTALL_FOLDER
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+SCRATCH_FOLDER=${SYS_ROOT}/"LIBGIT2_SCRATCH"
+LIBGIT2_SRC_FOLDER="${SCRATCH_FOLDER}/libgit2"
+LIBGIT2_PATCH_FOLDER="${SCRIPT_DIR}"/libgit2_patch
+LIBGIT2_BUILD_FOLDER="${SCRATCH_FOLDER}/libgit2-build"
+LIBGIT2_INSTALL_FOLDER="${SCRATCH_FOLDER}/libgit2-install"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+
+mkdir -p "${SCRATCH_FOLDER}"
+rm -rf "${SCRATCH_FOLDER:?}"/*
+
+pushd "${SCRATCH_FOLDER}"
+
+function libgit2_patch_build() {
+  pushd "${LIBGIT2_SRC_FOLDER}"
+  for patchfile in $(find -L "${LIBGIT2_PATCH_FOLDER}" -type f -name '*.patch' | sort); do
+    echo "Apply patch $patchfile..."
+    patch -p1 --quiet -i "$patchfile"
+  done
+  popd
+}
+
+function libgit2_build_shared() {
+  cmake -B "${LIBGIT2_BUILD_FOLDER}" -DBUILD_SHARED_LIBS=ON -DBUILD_TESTS=1 -DCMAKE_INSTALL_PREFIX="${LIBGIT2_INSTALL_FOLDER}" -DOPENSSL_ROOT_DIR="${AWS_LC_INSTALL_FOLDER}" -DUSE_SSH=exec -DUSE_HTTPS=openssl -DUSE_SHA1=HTTPS -DUSE_SHA256=HTTPS -DCMAKE_C_STANDARD=99 -DUSE_AUTH_NTLM=builtin
```

Comment:
should we specify `-DLINK_WITH_STATIC_LIBRARIES=0` to be explicit?

Reply:
I updated the script to set `LINK_WITH_STATIC_LIBRARIES` in either case.